File: python/texttest_fixture.py
# -*- coding: utf-8 -*-
from __future__ import print_function

from gilded_rose import *


def main():
    print("OMGHAI!")
    items = [
        generate_item(name="+5 Dexterity Vest", sell_in=10, quality=20),
        generate_item(name="Aged Brie", sell_in=2, quality=0),
        generate_item(name="Elixir of the Mongoose", sell_in=5, quality=7),
        generate_item(name="Sulfuras, Hand of Ragnaros", sell_in=0, quality=80),
        generate_item(name="Sulfuras, Hand of Ragnaros", sell_in=-1, quality=80),
        generate_item(name="Backstage passes to a TAFKAL80ETC concert", sell_in=15, quality=20),
        generate_item(name="Backstage passes to a TAFKAL80ETC concert", sell_in=10, quality=49),
        generate_item(name="Backstage passes to a TAFKAL80ETC concert", sell_in=5, quality=49),
        generate_item(name="Conjured Mana Cake", sell_in=3, quality=6),  # <-- :O
    ]
    days = 2
    import sys
    if len(sys.argv) > 1:
        days = int(sys.argv[1]) + 1
    for day in range(days):
        print("-------- day %s --------" % day)
        print("name, sellIn, quality")
        for item in items:
            print(item)
        print("")
        GildedRose(items).update_quality()


if __name__ == "__main__":
    main()

File: go/cmd/texttest_fixture.go
package cmd

import (
    "fmt"
    "os"
    "strconv"

    "github.com/emilybache/gildedrose-refactoring-kata/lib"
    "github.com/emilybache/gildedrose-refactoring-kata/models"
    "github.com/emilybache/gildedrose-refactoring-kata/domains"
    "github.com/spf13/cobra"
)

type TextTestFixtureCommand struct{}

func (this *TextTestFixtureCommand) Short() string {
    return "execute texttest fixture"
}

func (this *TextTestFixtureCommand) Setup(command *cobra.Command) {}

func (this *TextTestFixtureCommand) Run() lib.CommandRunner {
    return func(
        env                          lib.Env,
        logger                       lib.Logger,
        itemUpdateServiceProvider    domains.ItemUpdateServiceProvider,
    ) {
        fmt.Println("OMGHAI!")

        var itemModels = []*models.ItemModel{
            {"+5 Dexterity Vest", 10, 20},
            {"Aged Brie", 2, 0},
            {"Elixir of the Mongoose", 5, 7},
            {"Sulfuras, Hand of Ragnaros", 0, 80},
            {"Sulfuras, Hand of Ragnaros", -1, 80},
            {"Backstage passes to a TAFKAL80ETC concert", 15, 20},
            {"Backstage passes to a TAFKAL80ETC concert", 10, 49},
            {"Backstage passes to a TAFKAL80ETC concert", 5, 49},
            {"Conjured Mana Cake", 3, 6}, // <-- :O
        }

        var items = make([]*models.Item, len(itemModels))
        for i, item := range itemModels {
            items[i] = models.NewItem(item)
        }

        days := 2
        var err error
        if len(os.Args) > 2 {
            days, err = strconv.Atoi(os.Args[2])
            if err != nil {
                fmt.Println(err.Error())
                os.Exit(1)
            }
            days++
        }

        for day := 0; day < days; day++ {
            fmt.Printf("-------- day %d --------\n", day)
            fmt.Println("name, sellIn, quality")
            for i := 0; i < len(items); i++ {
                fmt.Println(items[i].Model)
            }
            fmt.Println("")
            for i := 0; i < len(items); i++ {
                itemUpdateService := itemUpdateServiceProvider.GetUpdateService(items[i])
                itemUpdateService.UpdateQuality(items[i])
            }
        }
    }
}

func NewTextTestFixtureCommand() *TextTestFixtureCommand {
    return &TextTestFixtureCommand{}
}
